### ISSUE TITLE
unable to understand the meaning of "otf_backprop_temperature". 

### ISSUE THREAD

**Issue by pmanocha023** [facebookresearch/UnsupervisedMT#55]
Hi,
I was looking at the code for "otf_backprop_temperature" and while doing back-translation and keeping the default setting(=-1), back-translating from lang1-lang2-lang3 would prevent the lossy translations from lang1-lang2 and just look into translations from lang2-lang3. With the default values, is this understanding correct?

```python
        if backprop_temperature == -1:
            # lang2 -> lang3
            encoded = self.encoder(sent2, len2, lang_id=lang2_id)
        else:
            # lang1 -> lang2
            encoded = self.encoder(sent1, len1, lang_id=lang1_id)
            scores = self.decoder(encoded, sent2[:-1], lang_id=lang2_id)
            assert scores.size() == (len2.max() - 1, bs, n_words2)

            # lang2 -> lang3
            bos = torch.cuda.FloatTensor(1, bs, n_words2).zero_()
            bos[0, :, params.bos_index[lang2_id]] = 1
            sent2_input = torch.cat([bos, F.softmax(scores / backprop_temperature, -1)], 0)
            encoded = self.encoder(sent2_input, len2, lang_id=lang2_id)
```

**Comment by glample** [comment]
Yes, your understanding is correct. I tried to back-propagate through the generated back-translation (`sent2`, the sentence associated to `lang2`), this was to reproduce the Cycle-GAN setting, but this didn't work well. In these experiments the back-translation loss became 0 very quickly, and the model was just behaving as the identity mapping. I only tried it once and did not tune the `backprop_temperature` value properly, but it's possible that a good tuning gives better results.
